apps/site/components/Common/Partners/index.tsx

```diff
@@ -0,0 +1,92 @@
+'use server';
+
+import Tooltip from '@node-core/ui-components/Common/Tooltip';
+import * as PartnerLogos from '@node-core/ui-components/Icons/PartnerLogos';
+
+import { providePartners } from '#site/next-data/providers/partners';
+import { partners as allPartners } from '#site/next.json.mjs';
+
+import type { Partner, PartnerCategory } from '#site/types';
+import type { FC } from 'react';
+
+import PartnerButton from './PartnerButton';
+
+import style from './index.module.css';
+
+type PartnersListProps = {
+  size?: 'big' | 'small';
+  category?: PartnerCategory;
+  sort?: 'name' | 'weight';
+  length?: number;
+};
+
+const getPartners = async (
+  length?: number,
+  category?: PartnerCategory,
+  sort?: 'name' | 'weight'
+) => {
+  let result: Array<Partner>;
+
+  if (sort === 'name') {
+    // For name sorting, just filter and sort directly - no need for weighted shuffling
+    const filtered = category
+      ? allPartners.filter(p => p.categories.includes(category))
+      : allPartners;
+    result = filtered.toSorted((a, b) => a.name.localeCompare(b.name));
+  } else {
+    // For weight sorting, use cached weighted partners
+    const minutes = 300; // Change every 5 minutes
+    const seed = Math.floor(Date.now() / (minutes * 1000));
+    result = await providePartners(seed, category);
+  }
+
+  return length ? result.slice(0, length) : result;
+};
```

Comment:
IMO `providePartners` should take no arguments. Each time the cache expires (every few minutes), it'll re-run, calculating the seed each time.

```suggestion
const getPartners = async (
  length?: number,
  category?: PartnerCategory,
  sort?: 'name' | 'weight'
) => {
  let result = sort === 'name' ? partners : await providePartners();
  
  result = category
      ? result.filter(p => p.categories.includes(category))
      : result;

  return length ? result.slice(0, length) : result;
};
```

Reply:
From what I understood from the documentation, the cache is invalidated based on the arguments, if they’re different, then it’s invalidated. 

How often does the cache expire here in Next.js?

Reply:
> How often does the cache expire here in Next.js?

5 minutes - 15 minutes